src/Transport/Topology/PublishersConfiguration.cs

```diff
@@ -0,0 +1,76 @@
+﻿namespace NServiceBus.AzureServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Linq;
+    using NServiceBus.AzureServiceBus.TypesScanner;
+    using NServiceBus.Settings;
+
+    class PublishersConfiguration
+    {
+        private readonly IConventions _conventions;
+        private readonly Dictionary<Type, List<string>> _publishers;
+
+        public PublishersConfiguration(IConventions conventions)
+        {
+            _conventions = conventions;
+            _publishers = new Dictionary<Type, List<string>>();
+        }
+
+        public void Map(string publisherName, Type type)
+        {
+            var types = type
+                .GetParentTypes()
+                .Union(new[] { type })
+                .Where(t => _conventions.IsMessageType(t))
+                .ToArray();
+
+            Array.ForEach(types, t => AddPublisherForType(publisherName, t));
+        }
+
+        public void Map(string publisherName, IEnumerable<Type> types)
+        {
+            foreach (var type in types)
+                Map(publisherName, type);
+        }
+
+        public IEnumerable<string> GetPublishersFor(Type type)
+        {
+            return new ReadOnlyCollection<string>(_publishers[type]);
+        }
+
+        public bool HasPublishersFor(Type type)
+        {
+            return _publishers.ContainsKey(type);
+        }
+
+        public static PublishersConfiguration ConfigureUsingThese(ReadOnlySettings settings)
```

Comment:
I get what then name does, but it's a little weird. 
`ConfigureWith` or `ConfigureUsing`?